core/crypto/vrf/impl/vrf_provider_impl.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/vrf/impl/vrf_provider_impl.hpp"
+
+#include "common/outcome.hpp"
+
+namespace fc::crypto::vrf {
+
+  VRFProviderImpl::VRFProviderImpl(
+      std::shared_ptr<bls::BlsProvider> bls_provider)
+      : bls_provider_(std::move(bls_provider)) {}
+
+  outcome::result<VRFResult> VRFProviderImpl::computeVRF(
+      const VRFSecretKey &secret_key, const VRFHash &msg) const {
+    auto &&result = bls_provider_->sign(msg, secret_key);
+    if (!result) {
+      return VRFError::SIGN_FAILED;
```

Comment:
Why not pass the original error? Can't it contain valuable information about the error?

Reply:
it is questionable, original error are not always acceptable